### PR TITLE
Building an API pt. 6 - lesson 2 (Update create_app & create test DB) updates

### PR DESCRIPTION
**Pull request by kelsey-steven-ada**
I would love feedback on the new lesson name, it feels a bit long but was struggling with how to shorten it and represent the contents. (I had to rename the lesson since it no longer touches on `python-dotenv`, that content will be moved to lesson 4 where we use it in `conftest`.)

Also apologies that the changes aren't side-by-side because of the file rename 😭

[Pt. 6 Asana ticket](https://app.asana.com/0/1205655776549324/1206697443678572/f)
[Pt. 6 Lesson 2 ticket](https://app.asana.com/0/1205655776549324/1207017851991897/f)

### CODE REVIEW

File: api-6-testing/setting-configs-and-creating-the-test-db.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+- Create the test database for `hello-books-api`
+
+We will complete the following steps:
+
+1. Create a `.env` file
+2. Prevent our `.env` from being committed by using a `.gitignore` file
+3. Create environment variables in our `.env` file:
+   - URL for a development database
+   - URL for a test database
+4. Create a test database with the matching name
```
Comment (yangashley) [inline api-6-testing/setting-configs-and-creating-the-test-db.md:20]:
```suggestion
4. Create a test database with the name that matches the one provided to our `.env` file
```

What do you think about specifying what the matching name is?

File: api-6-testing/setting-configs-and-creating-the-test-db.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+5. Refactor our `create_app` method to:
+   1. Check for a configuration dictionary
+   2. Read the correct database location from `.env`
```
Comment (yangashley) [inline api-6-testing/setting-configs-and-creating-the-test-db.md:23]: Did you mean to have these sub bullet points numbered or have them as circles like on line 18 & 19?

I realize that I might be making comments on content that aren't your changes.

File: api-6-testing/setting-configs-and-creating-the-test-db.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+## Create Our `.env` File
+
+It is a common practice to name our environment files `.env` and place them in the root directory. Later in this series we will be using a tool that requires this exact set up, so we should create the `.env` file in our project root either in our code editor, or by running
+
+```bash
+$ touch .env
+```
+
+in the project root directory.
```
Comment (yangashley) [inline api-6-testing/setting-configs-and-creating-the-test-db.md:58]: The way the words flow in the browser look kind of choppy to me. 
image

How would you feel about changing it like:

or by running the following command:
```bash
$ touch .env
```

File: api-6-testing/setting-configs-and-creating-the-test-db.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+### Adding `.env` to `.gitignore`
+
+To make Git ignore our `.env` file, we open `.gitignore` in our text editor and add `.env` to the file on its own line.
+
+In the case that we already have contents in our `.gitignore` file, we can confirm whether `.env` is listed by searching the file. `.env` only needs to be listed once.
+
+## Populate the Environment Variables
+
+`.env` files are made of lines of the form:
```
Comment (yangashley) [inline api-6-testing/setting-configs-and-creating-the-test-db.md:82]: How do you feel about rewording this sentence like:

`.env` files have the following format:

File: api-6-testing/setting-configs-and-creating-the-test-db.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+```bash
+VARIABLE_NAME=variable value
+```
+
+| <div style="min-width:175px;"> Piece of Code </div> | Notes |
+| --------------------------------------------------- | ----- |
+| `VARIABLE_NAME`                                     | Name of the environment variable to set. Replace `VARIABLE_NAME` with a good name for our environment variable. It's conventional to name environment variables with CAPITAL_CASING, just like constant variables. |
```
Comment (yangashley) [inline api-6-testing/setting-configs-and-creating-the-test-db.md:90]: what about changing "good name" to "descriptive name"

File: api-6-testing/setting-configs-and-creating-the-test-db.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+| `=`                                                 | The environment variable name is separated from its value with `=`. We usually _don't_ surround it with spaces. |
+| `variable value`                                    | The value for our environment variable. Surrounding quotes are optional in general, but may be required if we need our value to begin with spaces, since spaces immediately following the `=` are ignored. |
+
+Beyond those guidelines, we are free to choose the best variable names and values for our project.
```
Comment (yangashley) [inline api-6-testing/setting-configs-and-creating-the-test-db.md:94]:
```suggestion
Beyond those guidelines, we are free to choose variable names and values that are best suited for our project.
```

File: api-6-testing/setting-configs-and-creating-the-test-db.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+For our Hello Books API, we want to designate a path to a development database and a testing database with the variables named `SQLALCHEMY_DATABASE_URI` and `SQLALCHEMY_TEST_DATABASE_URI`.
```
Comment (yangashley) [inline api-6-testing/setting-configs-and-creating-the-test-db.md:96]:
```suggestion
For our Hello Books API, we want to designate separate paths to a development database and a testing database with the variables named `SQLALCHEMY_DATABASE_URI` and `SQLALCHEMY_TEST_DATABASE_URI`.
```

What do you think of this to add clarity that there are two different paths that can be taken.

File: api-6-testing/setting-configs-and-creating-the-test-db.md (hunk 1 above)
Comment (kelsey-steven-ada) [reply api-6-testing/setting-configs-and-creating-the-test-db.md:20]: I like being clear about the names matching, what do you think about:

```suggestion
4. Create a test database whose name matches the one provided in our `.env` file 
```

File: api-6-testing/setting-configs-and-creating-the-test-db.md (hunk 2 above)
Comment (kelsey-steven-ada) [reply api-6-testing/setting-configs-and-creating-the-test-db.md:23]: These actions have to happen in this order, so I think numbers are more fitting than bullets in this case.

File: api-6-testing/setting-configs-and-creating-the-test-db.md (hunk 5 above)
Comment (kelsey-steven-ada) [inline api-6-testing/setting-configs-and-creating-the-test-db.md:90]:
```suggestion
| `VARIABLE_NAME`                                     | Name of the environment variable to set. Replace `VARIABLE_NAME` with a descriptive name for our environment variable. It's conventional to name environment variables with CAPITAL_CASING, just like constant variables. |
```

File: api-6-testing/setting-configs-and-creating-the-test-db.md (hunk 4 above)
Comment (kelsey-steven-ada) [inline api-6-testing/setting-configs-and-creating-the-test-db.md:82]:
```suggestion
`.env` files typically use the following format:
```

File: api-6-testing/setting-configs-and-creating-the-test-db.md (hunk 1 above)
Comment (yangashley) [reply api-6-testing/setting-configs-and-creating-the-test-db.md:20]: Sounds good to me!

**Review by yangashley — APPROVED** [review]

**Pull request merged**